### PR TITLE
Added the interview creation feature

### PR DESCRIPTION
**Pull request by robin-khurana**
Added the basic functionalities in the interview creation app
-> Defined roles
-> User can schedule interview
-> User can add resume
-> Only admin can assign_interviewer or interviewee and changes status
-> Applied proper validations

ToDo
-> Adding mailers
-> Adding calender

### CODE REVIEW

**Comment by robin-khurana** [comment]
Screenshot 2020-05-19 at 9 55 37 AM
Screenshot 2020-05-19 at 9 56 04 AM
Screenshot 2020-05-19 at 9 59 43 AM

Admin view
Screenshot 2020-05-19 at 10 01 54 AM

User View
Screenshot 2020-05-19 at 9 59 59 AM

Autocomplete for assignment
Screenshot 2020-05-19 at 10 05 19 AM

File: app/views/interviews/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    <%= form.label :start_time %><br />
+    <%= form.time_field :start_time, required: true %> <br />
+  </div>
+  <div class="field">
+    <%= form.label :end_time %><br />
+    <%= form.time_field :end_time, required: true %> <br />
+  </div>
+  <div class="field"
+  <%= form.label :resume %> <br /><br/>
+  <%= form.file_field :attachment %> <br />
```
Comment (ayushg007) [inline app/views/interviews/_form.html.erb:20]: Can we have a validation on type of attachment ?

File: app/views/interviews/_form_head.html.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<thead>
+<tr>
+  <th>Title &nbsp; &nbsp;</th>
+  <th>Date &nbsp; &nbsp; </th>
+  <th>Start_Time &nbsp; &nbsp; </th>
+  <th>End_Time &nbsp; &nbsp; </th>
+  <th>Interviewee &nbsp; &nbsp; </th>
+  <th>Interviewer &nbsp; &nbsp; </th>
+  <th>Status &nbsp; &nbsp; </th>
```
Comment (ayushg007) [inline app/views/interviews/_form_head.html.erb:9]: not a good practice to use &nbsp, try to give margin instead

**Review by ayushg007 — COMMENTED** [review]
Hey Robin, Awesome work !!! added comments. 
One feedback frm me, u hv done a lot of extraa things like sending mails for password verification,authentication. That's great but I think ki we dont need all of that, but want other set of tasks like sending mails on interview update etc, so just try to complete the given task first thn go on enhancing ur app as much as u can.

**Comment by robin-khurana** [comment]
@ayushg007 Thanks for the feedback, I will complete the remaining tasks.

File: app/assets/javascripts/application.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+         function (data, status) {
+            if (status === 'success') {
+                alert("Status successfully updated")
+            }
+         }
+        )
+    });
+
+    fetch_auto_complete_info('/get_interview_users', "interviewee_select", "Enter interviewee name");
+    fetch_auto_complete_info('/get_interview_users', "interviewer_select", "Enter interviewer name");
```
Comment (jaindheerain) [inline app/assets/javascripts/application.js:39]: why we have same call twice?

**Pull request merged**